### ISSUE TITLE
False positive CS8619 warning using `-=` with `byte?`

### ISSUE THREAD

**Issue by jeremysalwen** [dotnet/roslyn#82552]
**Version Used**: 9.0.205

**Steps to Reproduce**: 

Compile a file a file containing
```
byte? b = 1;
b -= 1;
```

.NET Lab link: https://lab.razor.fyi/#41Li4gooyk8vSszVSy4WEkmqLEm1V0hSsFUwtOZKUtAF0V5MycVRnByXPn05NTldgCWBsYIRAA

**Diagnostic Id**:

Nullability of reference types in value of type 'int?' doesn't match target type 'byte?'.[CS8619](https://msdn.microsoft.com/query/roslyn.query?appId=roslyn&k=k(CS8619))

**Expected Behavior**:
No warning, since subtracting 1 from a nullable byte is fine.

**Actual Behavior**:
CS8619 warning appears. 

Note: Even if somehow there should be a warning since  `-=` should never be used with `byte?` types, there still is a bug here in that the warning message is incorrect, since the nullablility of `int?` *does* match the nullability of `byte?`.

**Comment by RikkiGibson** [comment]
This is a real bug. This kind of false warning tends to get reported because the particular spot in NullableWalker expected the types to only differ in nullable annotations, but instead there is a real type difference.
